nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -478,16 +507,20 @@ def _create_masks(self, max_audio_length, padding_length, device):
             padding_length.size(0), -1
         ) < padding_length.unsqueeze(-1)
 
-        # pad_mask_for_att_mask is the mask which helps to ignore paddings
-        pad_mask_for_att_mask = pad_mask.unsqueeze(1).repeat([1, max_audio_length, 1])
-        pad_mask_for_att_mask = torch.logical_and(pad_mask_for_att_mask, pad_mask_for_att_mask.transpose(1, 2))
-        # att_mask is the masking to be used by the MHA layers to ignore the tokens not supposed to be visible
-        att_mask = self.att_mask[:, :max_audio_length, :max_audio_length]
-        # paddings should also get ignored, so pad_mask_for_att_mask is used to ignore their corresponding scores
-        att_mask = torch.logical_and(pad_mask_for_att_mask, att_mask.to(pad_mask_for_att_mask.device))
+        if self.att_mask is not None:
+            # pad_mask_for_att_mask is the mask which helps to ignore paddings
+            pad_mask_for_att_mask = pad_mask.unsqueeze(1).repeat([1, max_audio_length, 1])
+            pad_mask_for_att_mask = torch.logical_and(pad_mask_for_att_mask, pad_mask_for_att_mask.transpose(1, 2))
+            # att_mask is the masking to be used by the MHA layers to ignore the tokens not supposed to be visible
+            att_mask = self.att_mask[:, :max_audio_length, :max_audio_length]
+            # paddings should also get ignored, so pad_mask_for_att_mask is used to ignore their corresponding scores
+            att_mask = torch.logical_and(pad_mask_for_att_mask, att_mask.to(pad_mask_for_att_mask.device))
+
+            att_mask = ~att_mask
+        else:
+            att_mask = None
 
         pad_mask = ~pad_mask
-        att_mask = ~att_mask
```

Comment:
How can att_mask be none??

Reply:
The att_mask isn't used for this approach, and in fact we can't even create it since we want to avoid quadratic memory. We do still have pad_mask

Reply:
Hmm it might cause problems for the middle downsampling but well deal with it later.